### PR TITLE
modified code

### PR DESCRIPTION
**Pull request by PritiSah6299**
Cleaned and modified code.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -1,77 +1,39 @@
...
-  
-  
-    let [todos, dispatch] = useReducer(reducer, [
+function App() { 
+    const [todos, dispatch] = useReducer(reducer, [
         { id: 1, text: 'demoText1', completed: false },
         { id: 2, text: 'demoText2', completed: false },
       ]);
 
     const [filter, setFilter] = useState('all');
```
Comment (kbirulia) [inline src/App.js:34]: Would be nice to have constants for filter values

File: src/App.js [hunk 2]
```diff
@@ -1,77 +1,39 @@
...
 
-    const filterTodos= todos.filter(todo => {
+    const filterTodos= useMemo(()=> {return todos.filter(todo => {
```
Comment (kbirulia) [inline src/App.js:36]: For performance reason is better to filter separetly for each filter (all is not needed to be filtered)

File: src/App.js [hunk 3]
```diff
@@ -83,79 +45,23 @@ function App() {
...
-
-  // const handleToggle = (id) => {
-  //   setTodos((prevTodos) =>
-  //     prevTodos.map((todo) =>
-  //       todo.id === id ? { ...todo, completed: !todo.completed } : todo
-  //     )
-  //   );
-  // };
-
   const inpst={
```
Comment (kbirulia) [inline src/App.js:53]: Use styled components outside of component

File: src/App.js [hunk 4]
```diff
@@ -83,79 +45,23 @@ function App() {
...
-//   if (event.key === 'Enter'){
-//   await setTodos((prevArray) => [...prevArray, {id:new Date().getTime()+Math.random()*100, text: event.target.value, completed: false}]);
-//   // event.target.value='';
-//   inputRef.current.value = '';
-//   }
-// };
-
 const handleInputChange = async(event) => {
   if (event.key === 'Enter'){
   await dispatch({type:'update',payload: event.target.value});
```
Comment (kbirulia) [inline src/App.js:59]: Would be nice to have constants for types

File: src/App.js [hunk 5]
```diff
@@ -166,9 +72,6 @@ const handleInputChange = async(event) => {
         placeholder='What needs to be done?'
         style={inpst}
         ref={inputRef}
-        // onFocus={handleFocus}
-        // value={inputValue}
-        // onKeyPress={handleInputChange}
         onKeyDown={handleInputChange}
         />
```
Comment (kbirulia) [inline src/App.js:77]: onToggle is not dispatch. Keep, please, naming meaningful

**Review by kbirulia — CHANGES_REQUESTED** [review]
Great job!

**Comment by PritiSah6299** [comment]
Need to add a delete feature for list.

**Pull request merged**